pytorch_lightning/trainer/trainer.py

```diff
@@ -1276,18 +1276,9 @@ def _call_setup_hook(self) -> None:
         self.accelerator.barrier("post_setup")
 
     def _call_configure_sharded_model(self) -> None:
-        # Call configure sharded model hook if accelerator requests. In some cases
-        # we will not call the hook; the hook has initialized the sharded model for example.
-
-        # used on the model if the user re-create a trainer with resume_from_checkpoint
-        model = self.lightning_module
-        model_call_configure_sharded_model_hook = getattr(model, "call_configure_sharded_model_hook", False)
-        if self.accelerator.call_configure_sharded_model_hook and not model_call_configure_sharded_model_hook:
-            with self.accelerator.model_sharded_context():
-                self.call_hook("configure_sharded_model")
-                self.call_hook("on_configure_sharded_model")
-            model.call_configure_sharded_model_hook = True
-            self.accelerator.call_configure_sharded_model_hook = False
+        with self.accelerator.model_sharded_context():
```

Comment:
what I found so far is that this call to the model_sharded_context is causing the issue if it gets called for the deepspeed plugin. If I add 

```python
        if isinstance(self.training_type_plugin, DeepSpeedPlugin):
            return
```
before the call here, I don't see unrelated tests failing. 
So, investigating `DeepSpeedPlugin.model_sharded_context()`

Reply:
@ananthsub possible that I found the test that causes this. Can you try to got to the test
`test_deepspeed_multigpu_stage_3_warns_resume_training` in` tests/plugins/test_deepspeed_plugin`
and flip the flag `special=False` to `special=True`
?